Promise/Promise.swift

```diff
@@ -12,10 +12,16 @@ import Foundation
 #endif
 
 public protocol ExecutionContext {
+    var queue: DispatchQueue { get }
     func execute(_ work: @escaping () -> Void)
 }
 
 extension DispatchQueue: ExecutionContext {
+
+    public var queue: DispatchQueue {
+        return self
+    }
+
```

Comment:
I worry about coupling execution contexts too much to DispatchQueues. I was thinking we should add a completion block to the execute function or the callFulfill/callReject functions. Would that make sense?

Reply:
I did the required changes, please check out the last commit. 😉